yas3fs/__init__.py

```diff
@@ -346,7 +350,7 @@ def delete(self, prop=None, wait_until_cleared_proplist = None):
                             remove_empty_dirs_for_file(etag_filename)
                 self.content = None # If not
                 self.update_size(True)
-                for p in self.props.keys():
+                for p in list(self.props.keys()):
```

Comment:
Is the conversion to list here necessary?

Reply:
Yes, because it modifies self.props in place, which causes an error in python 3

Reply:
Ah! Makes sense.